src/Core/src/Request.php

```diff
@@ -52,7 +57,7 @@ final class Request
      * @param array<string, string> $query
      * @param array<string, string> $headers
      */
-    public function __construct(string $method, string $uri, array $query, array $headers, RequestStream $body)
+    public function __construct(string $method, string $uri, array $query, array $headers, RequestStream $body, string $hostPrefix = '')
```

Comment:
Hmmm, this feels kinda wrong. Could we set the endpoint instead?

Reply:
We cannot. The Request object does not know the info that is injected by the client later